### PR TITLE
Add config commands

### LINKED ISSUE #37: Add config commands
With this new webman root config file for adding additional package repo sources, it could be nice to have a commands that help a user manage their config without having to manually edit the file.

### PR DESCRIPTION
**Pull request by jolheiser**
Closes #37 

`webman config` - Base command allows changing top-level config, currently only the refresh interval
`webman config remove` - Remove a package repository
`webman config add` - Add a package repository

Part of me wants to be conservative and change add/remove to something like `repo-add` or another subcommand `repo add`
But that would get a bit verbose, and I'm not sure it's worth the conservation currently.

### CODE REVIEW

**Comment by candrewlee14** [comment]
Wow, great work. 
I had an idea that maybe the `add` command should validate if the contents at the given repo URL actually match the expected format of a package repo. That could be a separate PR though.

**Comment by jolheiser** [comment]
> Wow, great work. I had an idea that maybe the `add` command should validate if the contents at the given repo URL actually match the expected format of a package repo. That could be a separate PR though.

That would make sense, and may as well pull it down anyways during the add command. 
I can either patch it into this PR or in another, up to you if you think that's easier to review.

File: cmd/config/config.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cfg, err := config.Load()
+		if err != nil {
+			return err
+		}
+
+		q := &survey.Input{
+			Message: "Refresh interval",
+			Help:    "How long before webman should refresh a package repository",
```
Comment (candrewlee14) [inline cmd/config/config.go:31]: Maybe give an example string here?
Looks you do validate the input after which is good, but an example would probably help.

File: cmd/config/remove/config_remove.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		}
+
+		var pkg string
+		if err := survey.AskOne(q, &pkg, survey.WithValidator(survey.Required)); err != nil {
+			return err
+		}
+
+		for idx, pkgRepo := range cfg.PkgRepos {
+			if pkgRepo.Name == pkg {
+				cfg.PkgRepos = append(cfg.PkgRepos[:idx], cfg.PkgRepos[idx+1:]...)
```
Comment (candrewlee14) [inline cmd/config/remove/config_remove.go:41]: If this idx is the last in `cfg.pkgRepo`, will this throw an error trying to access `cfg.PkgRepos[idx+1]`?

**Review by candrewlee14 — COMMENTED** [review]
Just a couple questions/comments:

File: cmd/config/remove/config_remove.go (hunk 2 above)
Comment (jolheiser) [reply cmd/config/remove/config_remove.go:41]: No, for this call it should gracefully accept the lower bound as being the same size as the slice. 
https://go.dev/play/p/WTs_mRoP7Az

Comment (candrewlee14) [reply cmd/config/remove/config_remove.go:41]: Gotcha, thanks for the playground link.

File: cmd/config/config.go (hunk 1 above)
Comment (jolheiser) [reply cmd/config/config.go]: Done in https://github.com/candrewlee14/webman/pull/38/commits/709dd5c5e5398804ca691435f8cc53f47c56693a

**Comment by jolheiser** [comment]
wrt checking format of repo, I would like to do that in another PR as I think it may require some other changes that may be nice to abstract out.

**Pull request merged**